include/aspect/material_model/rheology/fk_viscosity.h

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright (C) 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_material_model_rheology_fk_viscosity_h
+#define _aspect_material_model_rheology_fk_viscosity_h
+
+#include <aspect/global.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    namespace Rheology
+    {
+      template <int dim>
+      class FkViscosity : public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Constructor.
+           */
+          FkViscosity();
+
+          /**
+           * Declare the parameters this function takes through input files.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters from the parameter file.
+           */
+          void
+          parse_parameters (ParameterHandler &prm);
+
+          /**
+           * Compute the viscosity based on the Frank-Kamenetskii approximation.
+           */
+          double
+          compute_viscosity (const double ref_viscosity,
+                             const double temperature,
+                             const unsigned int composition) const;
+
+        private:
+
+          double reference_T;
+          /**
+           * List of E for viscosity calculation.
+           */
+          std::vector<double> fk_visc_E;
```

Comment:
```suggestion
          std::vector<double> viscosity_ratios_frank_kamenetskii;
```

Reply:
Out of curiosity, why is this term referred to as a "viscosity ratio", as described in the parameter description within the .cc file? Part of my confusion is that `E` is normally used in reference to an activation energy.

Reply:
also the `frank_kamenetskii` addition is not really necessary. This is inside the `FrankKamenetskii` plugin, so `viscosity_ratios` should be sufficiently clear.